core/src/test/java/com/datastax/oss/driver/internal/core/cql/DefaultBatchStatementTest.java

```diff
@@ -97,4 +101,37 @@ public void should_not_issue_log_warn_if_statement_have_no_consistency_level_set
 
     verify(logger.appender, times(0)).doAppend(logger.loggingEventCaptor.capture());
   }
+
+  @Test
+  public void should_inherit_lwt_status() {
+    // SELECT is not allowed in practice but is sufficient for unit testing
+    SimpleStatement simpleStatement =
+        SimpleStatement.builder("SELECT * FROM some_table WHERE a = ?").build();
+    BoundStatement lwtBoundStatement = mock(DefaultBoundStatement.class);
+    when(lwtBoundStatement.isLWT()).thenReturn(true);
+
+    // Without LWT statements added
+    BatchStatementBuilder batchStatementBuilder = new BatchStatementBuilder(BatchType.UNLOGGED);
+    batchStatementBuilder.addStatement(simpleStatement);
+    assertThat(batchStatementBuilder.build().isLWT()).isFalse();
+
+    // Check if implicitly set to true after adding LWT bound statement
+    batchStatementBuilder.addStatement(lwtBoundStatement);
+    assertThat(batchStatementBuilder.build().isLWT()).isTrue();
+
+    // Check if explicit set overrides implicit resolution
+    batchStatementBuilder.setIsLWT(false);
+    assertThat(batchStatementBuilder.build().isLWT()).isFalse();
+    batchStatementBuilder = new BatchStatementBuilder(BatchType.UNLOGGED);
+    batchStatementBuilder.addStatement(simpleStatement);
+    batchStatementBuilder.setIsLWT(true);
+    assertThat(batchStatementBuilder.build().isLWT()).isTrue();
+
+    assertThat(BatchStatement.newInstance(BatchType.UNLOGGED).isLWT()).isFalse();
+    assertThat(BatchStatement.newInstance(BatchType.LOGGED).isLWT()).isFalse();
+    assertThat(BatchStatement.newInstance(BatchType.COUNTER).isLWT()).isFalse();
+    assertThat(BatchStatement.newInstance(BatchType.UNLOGGED, lwtBoundStatement).isLWT()).isTrue();
+    assertThat(BatchStatement.newInstance(BatchType.LOGGED, lwtBoundStatement).isLWT()).isTrue();
+    assertThat(BatchStatement.newInstance(BatchType.COUNTER, lwtBoundStatement).isLWT()).isTrue();
+  }
```

Comment:
Could you also add integration test that creates and executes batch that contains both LWT and non-LWT statements?

Reply:
+ test that would check that resulted LWT batch get routed properly.

Reply:
Will do

Reply:
See if `LWTLoadBalancing#should_use_only_one_node_when_lwt_batch_detected` is sufficient. Basically the same as methods above but for batch statement with non-LWT and LWT query.